src/Catalog/PackageCatalogItemCreator.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.Metadata.Catalog.Helpers;
+using NuGet.Services.Metadata.Catalog.Persistence;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    public sealed class PackageCatalogItemCreator
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ILogger _logger;
+        private readonly IAzureStorage _storage;
```

Comment:
this is the first time in the V3 jobs we introduce a strong dependency on Azure. Up until now they could have functioned the same way on file system as well. Can we avoid this? Can we hide all the Azure specific functionality behind interfaces?

Reply:
`IAzureStorage` is an interface, and there's nothing in the interface that is Azure-specific except the presence of `Azure` in the name.  I could rename it to `IBlobStorage` if you prefer.

I could add file system support here, but I don't want to.  [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).  The goal of this PR is to improve V3 throughput for repository signing, and we don't need file system support here.  This new code path is opt-in only.  If there is a strong case to add file system support, we can add it to the backlog.